### PR TITLE
PLAT-8718 Add usage telemetry option

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

Add usage telemetry option

## Testing

Added unit tests to make sure the value is passed through.
Manually tested on iOS and Android

### CODE REVIEW

File: Source/BugsnagExampleEditor.Target.cs [hunk 1]
```diff
@@ -7,6 +7,14 @@ public class BugsnagExampleEditorTarget : TargetRules
...
 	public BugsnagExampleEditorTarget(TargetInfo Target) : base(Target)
 	{
+
+
+		if (Target.Platform == UnrealTargetPlatform.IOS || Target.Platform == UnrealTargetPlatform.Mac)
+		{
+			bOverrideBuildEnvironment = true;
+			AdditionalCompilerArguments = "-Wno-unused-but-set-variable";
+		}
+
```
Comment (nickdowell) [inline Source/BugsnagExampleEditor.Target.cs:17]: Don't think this should be committed.

File: Plugins/Bugsnag/Source/Bugsnag/Private/Apple/Specs/ApplePlatformConfiguration.spec.cpp [hunk 2]
```diff
@@ -180,11 +180,22 @@ void FApplePlatformConfigurationSpec::Define()
...
+
 					CocoaConfig = FApplePlatformConfiguration::Configuration(Configuration);
+
 					TEST_TRUE(CocoaConfig.telemetry == 0);
+
+					Configuration->SetTelemetry(EBugsnagTelemetryTypes::Usage);
+
+					CocoaConfig = FApplePlatformConfiguration::Configuration(Configuration);
+
+					TEST_TRUE(CocoaConfig.telemetry == BSGTelemetryUsage);
```
Comment (nickdowell) [inline Plugins/Bugsnag/Source/Bugsnag/Private/Apple/Specs/ApplePlatformConfiguration.spec.cpp:198]: Did you mean to leave all these blank lines in? Looks out of place next to the other test cases.

File: Source/BugsnagExampleEditor.Target.cs (hunk 1 above)
Comment (rich-bugsnag) [reply Source/BugsnagExampleEditor.Target.cs:17]: Removed

File: Plugins/Bugsnag/Source/Bugsnag/Private/Apple/Specs/ApplePlatformConfiguration.spec.cpp (hunk 2 above)
Comment (rich-bugsnag) [reply Plugins/Bugsnag/Source/Bugsnag/Private/Apple/Specs/ApplePlatformConfiguration.spec.cpp]: fixed

File: Source/BugsnagExampleEditor.Target.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@ public class BugsnagExampleEditorTarget : TargetRules
 {
 	public BugsnagExampleEditorTarget(TargetInfo Target) : base(Target)
 	{
+
```
Comment (nickdowell) [inline Source/BugsnagExampleEditor.Target.cs:10]: What's this for?

Comment (rich-bugsnag) [reply Source/BugsnagExampleEditor.Target.cs]: 🤠

**Review by nickdowell — APPROVED** [review]

**Pull request merged**